tests/run-tests

```diff
@@ -100,35 +96,67 @@ if [ ! -x "$prog_tick_cluster" ]; then
     exit 1
 fi
 
-# Run tick cluster
-_print "Launching tick-cluster..."
-$prog_tick_cluster --interface=127.0.0.1 $package_dir/node_modules/ringpop/main.js >/dev/null &
-pid_tickcluster=$!
-
-sleep 1
-if ! ps -p $pid_tickcluster &>/dev/null; then
-    echo "ERROR: Launching tick-cluster failed" >&2
-    exit 1
-fi
-
 # Clean up tick-cluster on exit
 trap _cleanup exit
 
-# Wait for convergence
-_print "Waiting for convergence..."
-sleep 5
-
 # Test setup
 ${0%/*}/npm-bin-link ../package.json >/dev/null || exit 1
+
 export PATH=$(_normalise_path "${0%/*}/../.bin"):$PATH
+export PROG_RINGPOP="$prog_ringpop"
+export PACKAGE_DIR="$package_dir"
+
+run() {
+	# Check test files exist
+	if ! ls $@ >/dev/null; then
+		echo "ERROR: Test files missing." >&2
+		exit 1
+	fi
+
+	local exit_code=0
+	# Run tick cluster
+	_print "Launching tick-cluster..."
+	$prog_tick_cluster --interface=127.0.0.1 --port 3000 $prog_ringpop >/dev/null &
+	pid_tickcluster=$!
+
+	sleep 1
+	if ! ps -p $pid_tickcluster &>/dev/null; then
+		echo "ERROR: Launching tick-cluster failed" >&2
+		exit 1
+	fi
+
+	# Wait for convergence
+	_print "Waiting for convergence..."
+	sleep 2
+
+	# Run the tests
+	_print "Running tests..."
+	cram $cram_opts $@ || exit_code=1
+
+	_cleanup
+
+	return $exit_code
+}
+
+declare test_result=0
+# Accept test files as arguments
+if [ $# -gt 0 ]; then
+    run $@ || test_result=1
```

Comment:
As @Motiejus has already mentioned, `"$@"` should generally always be quoted. `"$@"` basically means "expand all tokens and surround each token in quotes". `$@` unquoted is equivalent to `$*`, which will break if a filename has a space in it, for example.

```
dan-mbp-work:~ dan$ set -- "Foo Bar" "Baz"
dan-mbp-work:~ dan$ for arg in $@; do echo "arg: $arg"; done
arg: Foo
arg: Bar
arg: Baz
dan-mbp-work:~ dan$ for arg in "$@"; do echo "arg: $arg"; done
arg: Foo Bar
arg: Baz
dan-mbp-work:~ dan$ 
```

Welcome to bash!

![roflcopter](https://dans.im/gifs/roflcopter.gif)